src/volatile.rs

```diff
@@ -25,8 +25,59 @@ impl<T: Copy> Volatile<T> {
     pub fn new(value: T) -> Self {
         Self(value)
     }
+
+    #[inline]
+    fn self_ptr(&self) -> NonNull<T> {
+        NonNull::from(&self.0)
+    }
+
+    #[inline(always)]
+    pub fn read_volatile(&self) -> T {
+        unsafe { self.self_ptr().as_ptr().read_volatile() }
+    }
+
+    #[inline(always)]
+    pub fn write_volatile(&mut self, value: T) {
+        unsafe { self.self_ptr().as_ptr().write_volatile(value) }
+    }
```

Comment:
These methods don't make sense. If a memory location must be accessed through volatile reads or writes, e.g. because it is MMIO, then it's not sound to hold a reference to it, because the Rust memory model allows the compiler to dereference references whenever it wants. So the only case where these methods can be called is when `Volatile` is being used incorrectly. The intention of this type is that it is only ever referred to via a pointer, and read and written via the `VolatileReadable` and `VolatileWritable` traits, possibly via the `volread!` and `volwrite!` macros.